Source/SA/RFmx SpecAn/RFmxSpecAn.cs

```diff
@@ -44,33 +41,332 @@ public struct AutoLevelConfiguration
         {
             public bool AutoLevelReferenceLevel;
             public double AutoLevelMeasureTime_s;
+            public double AutoLevelBandwidth_Hz;
             public static AutoLevelConfiguration GetDefault()
             {
                 return new AutoLevelConfiguration
                 {
                     AutoLevelReferenceLevel = false,
-                    AutoLevelMeasureTime_s = 10e-3
+                    AutoLevelMeasureTime_s = 10e-3,
+                    AutoLevelBandwidth_Hz=20e6
                 };
             }
         }
+        public struct TxpConfiguration
+        {
+            public double MeasurementInterval_s;
+            public double Rbw_Hz;
+            public RFmxSpecAnMXTxpRbwFilterType RbwFilterType;
+            public double RrfAlpha;
+
+            public RFmxSpecAnMXTxpVbwFilterAutoBandwidth VbwAuto;
+            public double Vbw_Hz;
+            public double VbwToRbwRatio;
+
+            public RFmxSpecAnMXTxpThresholdEnabled ThresholdEnabled;
+            public double ThresholdLevel_dBordBm;
+            public RFmxSpecAnMXTxpThresholdType ThresholdType;

```

Comment:
These seem to me to be less often used parameters; hence, I would consider excluding them by default. @smooresni / @heyItsDale what do you think?

Reply:
Hey Michael, I would disagree on that. Since we need to consider the RBW setting with respect to the signal bandwidth and required sample rate for correct measurement shouldn't we make the other settings available?

Reply:
the new pull request, based on the latest master contains the change by removing VBW and threshold settings, I recommend to keep the filtertype and alpha at least to be able to switch filters from default like use RBW=signal bw with filter-none to have a correct txp measurement

Reply:
Yeah, to clarify I meant the VBW and threshold settings. I agree that RBW and sample rate are needed, and we can keep the filter settings. Thanks!